### DISCUSSION TITLE
How to use windicss to generate Atomic-CSS-like style.

### DISCUSSION THREAD

**Discussion by lcjnil** [windicss/windicss#513] [Q&A]
Windicss is good, but make all utlitity classes into html file is too messy. It would be fine if windicss can generate Atomic-CSS-like style, here is the example:

```HTML
<div class="btn">
    this is a btn
</div>

<style>
.btn {
    @apply font-bold py-2 px-4 rounded;  // use windicss directive
    some-other-style: some-other-value;
}
</style>
```

will be transformed to:

```HTML
<div class="btn font-bold py-2 px-4 rounded">
    this is a btn
</div>

<style>
.btn {
    some-other-style: some-other-value;
}
</style>
```

If we can use `@apply` directive in raw css, the HTML file would be very clean to read. And it will be transformed to regular windicss use in HTML, which make it super small in production.

**Comment by antfu** [comment]
https://windicss.org/posts/v30.html#alias-config

**Reply by lcjnil** [reply]
> https://windicss.org/posts/v30.html#alias-config

I want to make all my css to be transformed like demo, alias is hard to archive that
